### PR TITLE
fixed macos compilation and config parser issues

### PR DESCRIPTION
**Pull request by jjr840430**

### CODE REVIEW

**Comment by jjr840430** [comment]
this pull request is related with issue #273

File: test/temp.c [hunk 1]
```diff
@@ -28,7 +31,7 @@ int main(void)
 	if (fd > 0)
 		close(fd);
 
-#ifndef __APPLE__
+#if _GUN_SOURCE
```
Comment (ronaldtse) [inline test/temp.c]: Did you mean `__GNU_SOURCE`?

Comment (jjr840430) [inline test/temp.c]: I think _GNU_SOURCE is right way.

**Comment by jjr840430** [comment]
@ronaldtse I can't open the Jenkins result page.

**Comment by jjr840430** [comment]
@ronaldtse ./runtests.sh is working well on my mac.
Is it possible to get logs from Jenkins?

**Comment by ronaldtse** [comment]
@jjr840430 I confirm that the macOS Jenkins build is actually passing, I've fixed the problem and it just can't be "re-run" until there is an update of code here.

**Comment by ronaldtse** [comment]
@jjr840430 looks like Travis' macOS build had the following problem:

```sh
$ if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
error: RPC failed; curl 56 SSLRead() return error -36
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
Error: Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core failed!
```

**Comment by ronaldtse** [comment]
I'll merge this first and keep the Travis build issue separate. Thanks @jjr840430 !

**Pull request merged**